knowledge_repo/app/routes/web_editor.py

```diff
@@ -88,228 +66,176 @@ def gitless_drafts():
         query = (query.outerjoin(PostAuthorAssoc, PostAuthorAssoc.post_id == Post.id)
                       .filter(PostAuthorAssoc.user_id == g.user.id))
 
-    webposts = query.all()
-
-    return render_template("web_posts.html", posts=webposts)
+    return render_template("web_posts.html", posts=query.all())
 
 
 @blueprint.route('/posteditor', methods=['GET', 'POST'])
 @PageView.logged
 def post_editor():
-    """ Render the web post editor, either with the default settings
-        or if the post already exists, with what has been saved
-    """
-    post_id = request.args.get('post_id', None)
-
-    headers = defaultdict(lambda: None)
-    status = current_repo.PostStatus.DRAFT
-    markdown = None
-    comments = None
-    thumbnail = ''
-    if post_id and post_id != 'None':
-        webpost = (db_session.query(Post)
-                             .filter(Post.id == post_id)
-                             .first())
-        webpost_path = webpost.path
-        thumbnail = webpost.thumbnail
-        kp = current_repo.post(webpost_path)
-        status = current_repo.post_status(webpost_path)
+    """ Render the web post editor, either with the default values
+        or if the post already exists, with what has been saved """
+    path = request.args.get('path', None)
+
+    # set defaults
+    data = {'title': None,
+            'status': current_repo.PostStatus.DRAFT.value,
+            'repositories': current_repo.uri.keys(),
+            'markdown': None,
+            'comments': None,
+            'thumbnail': '',
+            'can_approve': 0,
+            'username': g.user.username,
+            'created_at': datetime.now(),
+            'updated_at': datetime.now()}
+
+    if path is not None and path in current_repo:
```

Comment:
I'd like to be more explicit in this case unless you have strong reason not to be?